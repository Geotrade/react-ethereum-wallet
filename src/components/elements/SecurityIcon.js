import makeBlockie from 'ethereum-blockies-base64';
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { updateTransactionToSend } from '../../actions/actions.js';

export const SecurityIcon = props => {
  const icon = makeBlockie(props.hash);
  let divStyle = {
    backgroundImage: 'url(' + icon + ')',
  };

  function updateToTransaction(e) {
    e.stopPropagation();
    updateTransactionToSend({
      name: 'to',
      value: props.hash,
    });
  }

  return (
    <React.Fragment>
      <span
        className={props.classes}
        title="This is a security icon.  If there were any change to the address, 
      the resulting icon would be a completely different one"
        src={icon}
        style={divStyle}
      >
        <img src={icon} style={divStyle} className="identicon-pixel" alt="" />
      </span>
      {props.type === 'transactionHref' ? (
        <Link
          to={{ pathname: '/send-from/' + props.hash }}
          title={props.hash}
          onClick={e => updateToTransaction(e)}
        >
          {props.hash}
        </Link>
      ) : (
        ''
      )}
    </React.Fragment>
  );
};

export default connect(
  null,
  { updateTransactionToSend }
)(SecurityIcon);
