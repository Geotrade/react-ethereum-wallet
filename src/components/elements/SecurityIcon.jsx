import makeBlockie from 'ethereum-blockies-base64';
import React from 'react';
import { Link } from 'react-router-dom';

export const SecurityIcon = props => {
  const icon = makeBlockie(props.hash);
  let divStyle = {
    backgroundImage: 'url(' + icon + ')',
  };
  return (
    <React.Fragment>
      <span
        className={props.classes}
        title="This is a security icon.  If there were any change to the address, 
      the resulting icon would be a completely different one"
        src={icon}
        style={divStyle}
      >
        <img src={icon} style={divStyle} className="identicon-pixel" alt="" />
      </span>
      {props.type === 'transactionHref' ? (
        <Link to={{ pathname: '/send/' + props.hash }} title="{props.hash}">
          {props.hash}
        </Link>
      ) : (
        ''
      )}
    </React.Fragment>
  );
};

export default SecurityIcon;
