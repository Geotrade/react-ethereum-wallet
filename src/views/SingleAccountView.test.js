import React from 'react';
import renderer from 'react-test-renderer';

import fetchMock from 'fetch-mock';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

import SecurityIcon from '../components/elements/SecurityIcon.js';

describe('title html', () => {
  afterEach(() => {
    fetchMock.restore();
  });

  it('renders title', () => {
    const tree = renderer
      .create(
        <div className="dapp-sticky-bar dapp-container">
          <SecurityIcon
            type="singleAccountView"
            classes="dapp-identicon"
            hash={'0x0000000000000000000000000000000000000000'}
          />
        </div>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('account information', () => {
  it('renders title', () => {
    const tree = renderer
      .create(
        <React.Fragment>
          <h1>
            <span>Account</span>
            <em className="edit-name">Account</em>
            <i className="edit-icon icon-pencil" />
          </h1>
          <h2 className="copyable-address">
            <i className="icon-key" title="Account" />
            <span>{'0x0000000000000000000000000000000000000000'}</span>
          </h2>
          <div className="clear" />
        </React.Fragment>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('renders sticky container', () => {
  it('renders title', () => {
    const tree = renderer
      .create(
        <div className="dapp-sticky-bar dapp-container">
          <SecurityIcon
            type="singleAccountView"
            classes="dapp-identicon"
            hash={'0x0000000000000000000000000000000000000000'}
          />
        </div>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
