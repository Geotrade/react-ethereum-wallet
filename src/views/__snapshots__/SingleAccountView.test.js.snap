// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account information renders title 1`] = `
Array [
  <h1>
    <span>
        Account
    </span>
    <em
        className="edit-name"
    >
        Account
    </em>
    <i
        className="edit-icon icon-pencil"
    />
</h1>,
  <h2
    className="copyable-address"
>
    <i
        className="icon-key"
        title="Account"
    />
    <span>
        0x0000000000000000000000000000000000000000
    </span>
</h2>,
  <div
    className="clear"
/>,
]
`;

exports[`title html renders title 1`] = `
Array [
  <h1>
    <span>
        Account
    </span>
    <em
        className="edit-name"
    >
        Account
    </em>
    <i
        className="edit-icon icon-pencil"
    />
</h1>,
  <h2
    className="copyable-address"
>
    <i
        className="icon-key"
        title="Account"
    />
    <span>
        0x0000000000000000000000000000000000000000
    </span>
</h2>,
  <div
    className="clear"
/>,
]
`;
